### PR TITLE
moved businesslogic from controller to service(Mentor and Student)

### LINKED ISSUE #184: Fix DisableMentor method
replace business logic to MentorService

### PR DESCRIPTION
**Pull request by Hlib-mats**
I moves business logic from mentor and student controller to mentor and student services(
methods from the service are called in controllers ). Also added methods to Service Interfaces

### CODE REVIEW

**Comment by Hlib-mats** [comment]
(*:*?:%?: чет этот гит сложный по началу

File: CharlieBackend.Api/Controllers/MentorsController.cs [hunk 1]
```diff
@@ -91,22 +91,8 @@ public async Task<ActionResult> PutMentor(long mentorId, [FromBody]UpdateMentorD
...
-            var isDisabled = await _accountService.DisableAccountAsync((long)accountId);
-
-            if (isDisabled)
-            {
-                return NoContent();
-            }
-
-            return StatusCode(500, "Error occurred while trying to disable mentor account.");
+            var disabledMentorModel = await _mentorService.DisableMentorAsync(id);
+            return disabledMentorModel.ToActionResult();
```
Comment (SerjKocherga) [inline CharlieBackend.Api/Controllers/MentorsController.cs:95]: Add a new line before return.

File: CharlieBackend.Api/Controllers/StudentsController.cs [hunk 2]
```diff
@@ -141,23 +141,9 @@ public async Task<ActionResult> PutStudent(long studentId, [FromBody]UpdateStude
         [HttpDelete("{id}")]
         public async Task<ActionResult> DisableStudent(long id)
         {
+            var disabledStudentModel = await _studentService.DisableStudentAsync(id);
+            return disabledStudentModel.ToActionResult();
```
Comment (SerjKocherga) [inline CharlieBackend.Api/Controllers/StudentsController.cs:145]: Add a new line before return.

File: CharlieBackend.Business/Services/MentorService.cs [hunk 3]
```diff
@@ -173,5 +173,30 @@ public async Task<MentorDto> GetMentorByIdAsync(long mentorId)
...
+
+        public async Task<Result<MentorDto>> DisableMentorAsync(long mentorId)
+        {
+
+            var accountId = await GetAccountId(mentorId);
+            if (accountId == null)
+            {
+                return Result<MentorDto>.GetError(ErrorCode.NotFound, "Unknown mentor id.");
+            }
+
```
Comment (SerjKocherga) [inline CharlieBackend.Business/Services/MentorService.cs:185]: Remove this empty line.

File: CharlieBackend.Business/Services/MentorService.cs [hunk 4]
```diff
@@ -173,5 +173,30 @@ public async Task<MentorDto> GetMentorByIdAsync(long mentorId)
...
+
+            var mentor = await GetMentorByAccountIdAsync((long)accountId);
+            var isActive = await _accountService.IsAccountActiveAsync(mentor.Email);
+            if ((bool)!isActive)
+            {
+                return Result<MentorDto>.GetError(ErrorCode.NotFound, "This account is already disabled.");
+            }
+
+
```
Comment (SerjKocherga) [inline CharlieBackend.Business/Services/MentorService.cs:194]: Remove this empty line.

File: CharlieBackend.Business/Services/StudentService.cs [hunk 5]
```diff
@@ -187,5 +187,30 @@ public async Task<StudentDto> GetStudentByEmailAsync(string email)
...
+            var student = await GetStudentByAccountIdAsync((long)accountId);
+            var isActive = await _accountService.IsAccountActiveAsync(student.Email);
+            if ((bool)!isActive)
+            {
+                return Result<StudentDto>.GetError(ErrorCode.NotFound, "This account is already disabled.");
+            }
+
+
+            var disabled = await _accountService.DisableAccountAsync((long)accountId);
+            return Result<StudentDto>.GetSuccess(student);
```
Comment (SerjKocherga) [inline CharlieBackend.Business/Services/StudentService.cs:211]: Add a new line before return.

File: CharlieBackend.Business/Services/StudentService.cs [hunk 6]
```diff
@@ -187,5 +187,30 @@ public async Task<StudentDto> GetStudentByEmailAsync(string email)
...
+
+
+        }
     }
 }
```
Comment (SerjKocherga) [inline CharlieBackend.Business/Services/StudentService.cs:216]: Add a new line to the end of file.

File: CharlieBackend.Business/Services/MentorService.cs [hunk 7]
```diff
@@ -173,5 +173,30 @@ public async Task<MentorDto> GetMentorByIdAsync(long mentorId)
...
+            var disabled = await _accountService.DisableAccountAsync((long)accountId);
+            return Result<MentorDto>.GetSuccess(mentor);
+
```
Comment (SerjKocherga) [inline CharlieBackend.Business/Services/MentorService.cs:197]: Remove these empty lines after return.

File: CharlieBackend.Business/Services/MentorService.cs [hunk 8]
```diff
@@ -173,5 +173,30 @@ public async Task<MentorDto> GetMentorByIdAsync(long mentorId)
...
+            var mentor = await GetMentorByAccountIdAsync((long)accountId);
+            var isActive = await _accountService.IsAccountActiveAsync(mentor.Email);
+            if ((bool)!isActive)
+            {
+                return Result<MentorDto>.GetError(ErrorCode.NotFound, "This account is already disabled.");
+            }
+
+
+            var disabled = await _accountService.DisableAccountAsync((long)accountId);
+            return Result<MentorDto>.GetSuccess(mentor);
```
Comment (SerjKocherga) [inline CharlieBackend.Business/Services/MentorService.cs:196]: Add a new line before return.

File: CharlieBackend.Business/Services/MentorService.cs [hunk 9]
```diff
@@ -173,5 +173,30 @@ public async Task<MentorDto> GetMentorByIdAsync(long mentorId)
 
             return mentor?.AccountId;
         }
+
+        public async Task<Result<MentorDto>> DisableMentorAsync(long mentorId)
+        {
+
```
Comment (SerjKocherga) [inline CharlieBackend.Business/Services/MentorService.cs:179]: Remove this empty line.

File: CharlieBackend.Business/Services/StudentService.cs [hunk 10]
```diff
@@ -187,5 +187,30 @@ public async Task<StudentDto> GetStudentByEmailAsync(string email)
...
+        public async Task<Result<StudentDto>> DisableStudentAsync(long id)
+        {
+
+            var accountId = await GetAccountId(id);
+
+            if (accountId == null)
+            {
+                return Result<StudentDto>.GetError(ErrorCode.NotFound, "Unknown student id.");
+            }
+
```
Comment (SerjKocherga) [inline CharlieBackend.Business/Services/StudentService.cs:200]: Remove this empty line.

File: CharlieBackend.Business/Services/StudentService.cs [hunk 11]
```diff
@@ -187,5 +187,30 @@ public async Task<StudentDto> GetStudentByEmailAsync(string email)
...
+
+            var student = await GetStudentByAccountIdAsync((long)accountId);
+            var isActive = await _accountService.IsAccountActiveAsync(student.Email);
+            if ((bool)!isActive)
```
Comment (SerjKocherga) [inline CharlieBackend.Business/Services/StudentService.cs:204]: Add a new line before if statement.

**Review by SerjKocherga — CHANGES_REQUESTED** [review]

**Comment by Hlib-mats** [comment]
Fixed empty lines and added missing

**Review by SerjKocherga — APPROVED** [review]

File: CharlieBackend.Api/Controllers/MentorsController.cs [hunk 12]
```diff
@@ -91,22 +91,9 @@ public async Task<ActionResult> PutMentor(long mentorId, [FromBody]UpdateMentorD
         [HttpDelete("{id}")]
         public async Task<ActionResult> DisableMentor(long id)
         {
+            var disabledMentorModel = await _mentorService.DisableMentorAsync(id);
```
Comment (x1mx1x) [inline CharlieBackend.Api/Controllers/MentorsController.cs:94]: you can name variable just DisabledMentor
but it isn't required

File: CharlieBackend.Business/Services/StudentService.cs [hunk 13]
```diff
@@ -187,5 +187,27 @@ public async Task<StudentDto> GetStudentByEmailAsync(string email)
...
+
+            var student = await GetStudentByAccountIdAsync((long)accountId);
+            var isActive = await _accountService.IsAccountActiveAsync(student.Email);
+            
+            if ((bool)!isActive)
+            {
+                return Result<StudentDto>.GetError(ErrorCode.NotFound, "This account is already disabled.");
+            }
+
+            var disabled = await _accountService.DisableAccountAsync((long)accountId);
```
Comment (x1mx1x) [inline CharlieBackend.Business/Services/StudentService.cs:208]: it's uneccessary to create "disabled" variable. You won't use it

File: CharlieBackend.Business/Services/StudentService.cs [hunk 14]
```diff
@@ -187,5 +187,27 @@ public async Task<StudentDto> GetStudentByEmailAsync(string email)
...
+            
+            return Result<StudentDto>.GetSuccess(student);
+        }
     }
 }
```
Comment (x1mx1x) [inline CharlieBackend.Business/Services/StudentService.cs:213]: Add a new line to the end of the file.

**Review by x1mx1x — CHANGES_REQUESTED** [review]

File: CharlieBackend.Business/Services/StudentService.cs (hunk 13 above)
Comment (Hlib-mats) [reply CharlieBackend.Business/Services/StudentService.cs:208]: it will be changed in next pr, check fix all services

**Pull request closed without merging**